spec/support/command_helpers.rb

```diff
@@ -1,17 +1,227 @@
 # frozen_string_literal: true
 
-module CommandHelpers
-  def command_output
-    tmp_stderr = Tempfile.create
+require "securerandom"
+require "pty"
+require "expect"
 
-    allow_any_instance_of(Command::Base).to receive(:progress).and_return(tmp_stderr) # rubocop:disable RSpec/AnyInstance
+class SpawnedCommand
+  attr_reader :output, :input, :pid
 
-    yield
+  DEFAULT_TIMEOUT = 120
 
-    tmp_stderr.rewind
-    output = tmp_stderr.read
-    tmp_stderr.close
+  def initialize(output, input, pid)
+    @output = output
+    @input = input
+    @pid = pid
+  end
+
+  def wait_for(regex, timeout: DEFAULT_TIMEOUT)
+    result = nil
+    output.expect(regex, timeout) do |matches|
+      result = matches&.first
+    end
+
+    raise "Timed out waiting for #{regex.inspect} after #{timeout} seconds" if result.nil?
+
+    result
+  end
+
+  def wait_for_prompt
+    wait_for(/[$#>]/)
+  end
+
+  def type(string)
+    input.puts("#{string}\n")
+  end
+
+  def kill
+    Process.kill("INT", pid)
+  end
+end
+
+module CommandHelpers # rubocop:disable Metrics/ModuleLength
+  module_function
+
+  DUMMY_TEST_ORG = "shakacode-heroku-to-control-plane-ci"
+  DUMMY_TEST_APP_PREFIX = "dummy-test"
+  LOG_FILE = "spec.log"
+
+  def dummy_test_org
+    DUMMY_TEST_ORG
+  end
+
+  def dummy_test_app_prefix(extra_prefix = "")
+    prefix = DUMMY_TEST_APP_PREFIX
+    prefix += "-#{extra_prefix}" unless extra_prefix.nil? || extra_prefix.empty?
+
+    prefix
+  end
+
+  def dummy_test_app(extra_prefix = "", suffix = "")
+    prefix = dummy_test_app_prefix(extra_prefix)
+    suffix = SecureRandom.hex(4) if suffix.nil? || suffix.empty?
+
+    "#{prefix}-#{suffix}"
+  end
+
+  def create_app_if_not_exists(extra_prefix = "", suffix = "") # rubocop:disable Metrics/MethodLength, Metrics/PerceivedComplexity
+    app = if !extra_prefix.nil? && !extra_prefix.empty?
+            dummy_test_app(extra_prefix, "unique")
+          else
+            dummy_test_app(extra_prefix, suffix)
+          end
+
+    result = run_command("exists", "-a", app)
+    return app unless result[:status] == 1
```

Comment:
Can we log a message if exists or not, maybe if some ENV like VERBOSE_TESTS=='true' for debugging?